### PR TITLE
Move validity masking out from function to transform

### PR DESCRIPTION
**Pull request by junli1026**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datavalues/src/columns/data_column.rs [hunk 1]
```diff
@@ -210,6 +261,135 @@ impl DataColumn {
...
+        let bitmap_to_apply = match validity_to_apply {
+            DataColumnValidity::Array(v, _) => v,
+
+            // The validity is neither all-true nor all-false, it must NOT be a constant one.
+            DataColumnValidity::Constant(_, _) => unreachable!(),
+        };
+
+        match self {
+            DataColumn::Array(_) => {
+                let array_ref = self.get_array_ref()?;
```
Comment (sundy-li) [inline common/datavalues/src/columns/data_column.rs:319]: just use `array.validity()`

File: common/datavalues/src/columns/data_column.rs [hunk 2]
```diff
@@ -210,6 +261,135 @@ impl DataColumn {
...
+        for (i, v) in validities.iter().enumerate() {
+            if v.len() != array_len {
+                return Err(ErrorCode::BadArguments(format!(
+                    "Validity's length {} doesn't match column array len {}",
+                    v.len(),
+                    array_len
+                )));
+            }
+            if v.all_null() {
+                all_null_index = i as i32;
```
Comment (sundy-li) [inline common/datavalues/src/columns/data_column.rs:361]: so we can `return Ok(DataColumnValidity::Constant(false, array_len));` here.

all_null_index is useless.

File: common/datavalues/src/columns/data_column.rs [hunk 3]
```diff
@@ -210,6 +261,135 @@ impl DataColumn {
...
+            }
+            if !v.is_constant() {
+                bitmap_vec.push(v);
```
Comment (sundy-li) [inline common/datavalues/src/columns/data_column.rs:364]: So bitmap_vec can be Vec<Option<&Bitmap>> instead.  We don't need `unreachable` later.

File: common/functions/src/scalars/strings/concat.rs [hunk 4]
```diff
@@ -140,6 +140,10 @@ impl Function for ConcatFunction {
         let result = Self::concat_column(columns[0].column().clone(), &columns[1..])?;
         Ok(result)
     }
+
+    fn passthrough_null(&self) -> bool {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/concat.rs:144]: why this is false?

**Comment by BohuTANG** [comment]
Hello, any progress of this PR? @sundy-li @junli1026

**Comment by junli1026** [comment]
> Hello, any progress of this PR? @sundy-li @junli1026

Just had a call with Sundy, will update the PR soon.

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
:+1: , Thank you @junli1026 !

**Pull request merged**